### PR TITLE
Certhandler int tests

### PR DESCRIPTION
**Pull request by mykola-kobets-epam**

### CODE REVIEW

File: tests/include/log.hpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
```
Comment (al1img) [inline tests/include/log.hpp:1]: 1. Move mocks folder to `tests/iam`
2. We don't need separate `include` dir, all headers should be with corresponding cpp files.
3. Move `log.hpp` to utils.

File: tests/include/softhsmenv.hpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_SOFTHSMEnv_H_
```
Comment (al1img) [inline tests/include/softhsmenv.hpp:8]: move to utils/softhsm

Comment (al1img) [inline tests/include/softhsmenv.hpp:8]: `AOS_SOFTHSMENV_HPP_`

File: tests/include/softhsmenv.hpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+#define AOS_SOFTHSMEnv_H_
+
+#include "aos/common/pkcs11/pkcs11.hpp"
+
+namespace aos {
+namespace test {
+
+class SOFTHSMEnv {
```
Comment (al1img) [inline tests/include/softhsmenv.hpp:16]: `SoftHSMEnv`

File: tests/include/log.hpp (hunk 1 above)
Comment (mykola-kobets-epam) [reply tests/include/log.hpp:1]: 1. As the new integration tests include all test scenarious from the old unit tests I removed the latter together with required mocks.
2/3. done

File: tests/iam/certhandler/certhandler_integr_test.cpp [hunk 4]
```diff
@@ -0,0 +1,639 @@
...
+namespace iam {
+namespace certhandler {
+
+using namespace testing;
+
+/***********************************************************************************************************************
+ * Suite
+ **********************************************************************************************************************/
+
+class CertHandlerTest : public Test {
```
[the remark this answers is not in the document]
Comment (mykola-kobets-epam) [reply tests/iam/certhandler/certhandler_integr_test.cpp:28]: No, I decided to replace it. Doesn't make much sense to maintain both. The current one is renamed to iam_tests, as basically they test whole iam component including pkcs11certmodule.

File: tests/iam/certhandler/storagestub.hpp [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+#include "aos/iam/certmodules/certmodule.hpp"
+
+namespace aos {
+namespace iam {
+namespace certhandler {
+/**
+ * Stoage stub.
+ */
+class StorageStub : public StorageItf {
```
[the remark this answers is not in the document]
Comment (mykola-kobets-epam) [reply tests/iam/certhandler/storagestub.hpp:19]: As all the mocks were removed only StorageStub left.

File: tests/include/softhsmenv.hpp (hunk 3 above)
Comment (mykola-kobets-epam) [reply tests/include/softhsmenv.hpp:16]: fixed

**Review by al1img — APPROVED** [review]

**Review by MykolaSolyanko — APPROVED** [review]

**Pull request merged**